File: frontend/src/types/ai-advisor.ts
export interface AIPrediction {
  id: number
  symbol: string
  prediction_type: '24h' | 'week' | 'month' | 'year'
  predicted_price: number
  confidence_percent: number
  prediction_reasoning?: string
  model_name: string
  created_at: string
  is_verified: boolean
  actual_price_at_target?: number
  accuracy_percent?: number
}

export interface PredictionResponse {
  symbol: string
  predictions: {
    '24h'?: PredictionData
    week?: PredictionData
    month?: PredictionData
    year?: PredictionData
  }
}

export interface PredictionData {
  predicted_price: number
  confidence_percent: number
  reasoning?: string
  model_name?: string
  created_at?: string
  is_verified?: boolean
}

export interface NewsAnalysis {
  id: number
  symbol: string
  news_date: string
  title: string
  summary?: string
  sentiment_score?: number
  relevance_score?: number
  source?: string
  created_at: string
}

export interface ChartDataPoint {
  timestamp: number
  price: number
  date: string
}

export interface ChartData {
  symbol: string
  data: ChartDataPoint[]
}

export interface PerformanceStats {
  total_predictions: number
  average_accuracy: number
  by_model: Record<string, { count: number; avg_accuracy: number }>
  by_symbol: Record<string, { count: number; avg_accuracy: number }>
}

File: backend/app/schemas/ai_advisor.py
from typing import Optional, Dict, Any
from pydantic import BaseModel, ConfigDict


class AIPrediction(BaseModel):
    model_config = ConfigDict(protected_namespaces=())

    id: int
    symbol: str
    prediction_type: str
    predicted_price: float
    confidence_percent: float
    prediction_reasoning: Optional[str] = None
    model_name: str
    created_at: str
    is_verified: bool = False
    actual_price_at_target: Optional[float] = None
    accuracy_percent: Optional[float] = None


class PredictionResponse(BaseModel):
    """Response format for predictions"""
    symbol: str
    predictions: Dict[str, Dict[str, Any]]

    class Config:
        json_encoders = {
            float: lambda v: round(v, 8) if v is not None else None
        }


class NewsAnalysis(BaseModel):
    id: int
    symbol: str
    news_date: str
    title: str
    summary: Optional[str] = None
    sentiment_score: Optional[float] = None
    relevance_score: Optional[float] = None
    source: Optional[str] = None
    created_at: str


class ChartDataPoint(BaseModel):
    timestamp: int
    price: float
    date: str


class ChartData(BaseModel):
    symbol: str
    data: list[ChartDataPoint]


class PerformanceStats(BaseModel):
    total_predictions: int
    average_accuracy: float
    by_model: Dict[str, Dict[str, Any]]
    by_symbol: Dict[str, Dict[str, Any]]


class PredictionRequest(BaseModel):
    symbol: str
    force_regenerate: bool = False
